prod/progress_bar/gadget.xml

```diff
@@ -0,0 +1,396 @@
+            noUiSlider.create(progressSlider, {
+                start: 0,
+                connect: [true, false],
+                step: 1,
+                tooltips: true,
+                range: {
+                  'min': 0,
+                  'max': 100
+                },
+                format: {
+                  to: function ( value ) {
+                    return value + '%';
+                  },
+                  from: function ( value ) {
+                    return value.replace('%', '');
+                  }
+                }
+            });
+            progressSlider.setAttribute('disabled', true);
+
+            var metadata;
+            var actionLogger;
+            var vaultFile;
+            var loadingStudents = false;
+
+            var getStudentPercentages = function () {
+                loadingStudents = true;
+                ils.filterVault(metadata.storageId, null, metadata.generator.id, null, null, null, null, null, function (resources) {
+                    if (resources.err) {
+                        loadingStudents = false;
+                        return console.error(resources.err);
+                    }
+
+                    teacherViewLoaded = true;
+                    if (resources.length) {
+                        resources.sort(function(a,b){
+                            return a.metadata.actor.displayName.toUpperCase().localeCompare(b.metadata.actor.displayName.toUpperCase());
+                        });
+                        $.each(resources, function(index, vaultFile) {
+                            if(vaultFile.metadata.actor.objectType == "graasp_student") {
+                                var displayName = vaultFile.metadata.actor.displayName;
+                                var actorId = vaultFile.metadata.actor.id;
+                                var date = new Date (vaultFile.updated);
+                                date = date.toLocaleString('en-GB');
+                                var progress = "";
+                                try {
+                                   progress = JSON.parse(vaultFile.content).progress;
+                                } catch (e) {
+                                    console.error(e);
+                                }
+                                $('table tbody').append('<tr><td>'+displayName+'</td><td>'+progress+'%</td><td>'+date+'</td></tr>');
+                            }
+                        });
+                        if (spinTimeout) clearTimeout(spinTimeout);
+                        $('.loader').remove();
+                        gadgets.window.adjustHeight();
+                        loadingStudents = false;
+                    }
+                });
+            };
+
+            $('.refresh-button').click(function() {
+                if (loadingStudents) return;
+                $('table tbody').empty();
+                $('.teacher-content').append('<p></p><div class="loader"></div>');
```

Comment:
I wouldn't add these `<p></p>` elements otherwise they will keep building up each time the loader is displayed. (You're only calling `$('.loader').remove();`)